Add google authentication to Jenkins.

MaximSternberg opened the pull request:
Added new variables to jenkins-dev.yaml.

-   google_oauth_client_id
-   google_oauth_client_secret
-   clients_rbac

Format of clients_rbac
```
 - Jenkins Role:
   - email
```

For example:
```
  clients_rbac:
    - 'Overall/Administer':
      - 'user1@example.com'
    - 'Overall/Read'
      - 'user2@example.com'
      - 'user3@example.com'
```

Added to jenkins.yaml new plugin - **google-login-plugin**
and also was deleted default admin password, because we can't use both of authentications at the same time.

In jenkins.yaml.erb was changed '**securityRealm**' and '**authorizationStrategy**'

	changed:      data/role/jenkins-dev.yaml
	changed:      data/role/jenkins.yaml
	changed:      templates/jenkins.yaml.erb

alvagante commented on templates/jenkins.yaml.erb:
```diff
@@ -8,13 +8,24 @@ plugins:
+    grantedPermissions:
+<% @options_hash['clients_rbac'].each do |roles_list| -%>
+<% roles_list.each do |users_list| -%>
+<% users_list[1].each do |user| -%>
```
Have you tested that the file content has exactly what you expect given the existing data?
At first look it seems ok anyway

olegy-ait commented:
@antgel Need to set appropriate values for: 
google_oauth_client_id - need to create on Google account
google_oauth_client_secret - need to create on Google account
clients_rbac - Which users and which roles.

Please provide it.

antgel commented:
@olegy-ait Where? This is a public repo.

antgel commented on data/role/jenkins-dev.yaml:
```diff
@@ -9,6 +9,13 @@ psick::jenkins::jcasc::options_hash:
+    - 'Overall/Administer':
+      - 'maxim.sternberg@automat-it.com'
+      - 'oleg.yrchenko@automat-it.com'
+      - 'antony.gelberg@graduway.com'
```
Please add al@example42.com.

MaximSternberg replied:
> Please add [al@example42.com](mailto:al@example42.com).
Done

MaximSternberg committed:
Added new variables to jenkins-dev.yaml.

google_oauth_client_id
google_oauth_client_secret
clients_rbac
Format of clients_rbac

 - Jenkins Role:
   - email
For example:

  clients_rbac:
    - 'Overall/Administer':
      - 'user1@example.com'
    - 'Overall/Read'
      - 'user2@example.com'
      - 'user3@example.com'
Added to jenkins.yaml new plugin - google-login-plugin
and also was deleted default admin password, because we can't use both of authentications at the same time.

In jenkins.yaml.erb was changed 'securityRealm' and 'authorizationStrategy'

changed:      data/role/jenkins-dev.yaml
changed:      data/role/jenkins.yaml
changed:      templates/jenkins.yaml.erb
